redwood-gradle-plugin/src/main/kotlin/app/cash/redwood/gradle/RedwoodSchemaPlugin.kt

```diff
@@ -52,22 +57,54 @@ public class RedwoodSchemaPlugin : Plugin<Project> {
     val compilation = kotlin.target.compilations.getByName(MAIN_COMPILATION_NAME)
     val classpath = project.configurations.getByName(compilation.compileDependencyConfigurationName)
 
-    val generate = project.tasks.register("redwoodGenerate", RedwoodSchemaTask::class.java) {
-      it.group = LifecycleBasePlugin.BUILD_GROUP
+    val generateJson = project.tasks.register("redwoodJsonGenerate", RedwoodSchemaJsonTask::class.java) {
+      it.group = BUILD_GROUP
       it.description = "Generate parsed schema JSON"
 
       it.toolClasspath.from(toolingConfiguration)
       it.outputDir.set(project.layout.buildDirectory.dir("generated/redwood"))
       it.schemaType.set(extension.type)
       it.classpath.from(classpath, compilation.output.classesDirs)
     }
+    project.tasks.named(JAR_TASK_NAME, Jar::class.java).configure {
+      it.from(generateJson)
+    }
+
+    project.afterEvaluate {
```

Comment:
Note or TODO on why the naughty afterEvaluate is needed? I assume it’s related to `compilation` not being ready?

Reply:
`afterEvaluate` allows the user's script to run and set the boolean which controls whether these tasks get created or not. The alternative is to always create the tasks and have them no-op if the boolean is set.

Reply:
Commented